v2/pkg/engine/datasource/graphql_datasource/representation_variable_test.go

```diff
@@ -193,50 +193,255 @@ func TestBuildRepresentationVariableNode(t *testing.T) {
+					OnTypeNames: [][]byte{[]byte("User")},
+				},
+			},
+		}
+
+		merged := mergeRepresentationVariableNodes([]*resolve.Object{userKeyRepresentation, userRequiresRepresentation})
+		require.Equal(t, expected, merged)
+	})
+
+	t.Run("same entity deeply nested fields", func(t *testing.T) {
+		userKeyRepresentation := &resolve.Object{
+			Fields: []*resolve.Field{
+				{
+					Name: []byte("id"),
+					Value: &resolve.String{
+						Path: []string{"id"},
+					},
+					OnTypeNames: [][]byte{[]byte("User")},
+				},
+				{
+					Name: []byte("userInfo"),
+					Value: &resolve.Object{
+						Path: []string{"userInfo"},
+						Fields: []*resolve.Field{
+							{
+								Name: []byte("kind"),
+								Value: &resolve.String{
+									Path: []string{"kind"},
+								},
+							},
+							{
+								Name: []byte("addresses"),
+								Value: &resolve.Array{
+									Path: []string{"addresses"},
+									Item: &resolve.Object{
+										Fields: []*resolve.Field{
+											{
+												Name: []byte("zip"),
+												Value: &resolve.String{
+													Path: []string{"zip"},
+												},
+											},
+										},
+									},
+								},
+							},
+						},
+					},
+					OnTypeNames: [][]byte{[]byte("User")},
+				},
+			},
+		}
+
+		userRequiresRepresentation := &resolve.Object{
+			Fields: []*resolve.Field{
+				{
+					Name: []byte("name"),
+					Value: &resolve.String{
+						Path: []string{"name"},
+					},
+					OnTypeNames: [][]byte{[]byte("User")},
+				},
+				{
+					Name: []byte("userInfo"),
+					Value: &resolve.Object{
+						Path: []string{"userInfo"},
+						Fields: []*resolve.Field{
+							{
+								Name: []byte("type"),
+								Value: &resolve.String{
+									Path: []string{"type"},
+								},
+							},
+							{
+								Name: []byte("addresses"),
+								Value: &resolve.Array{
+									Path: []string{"addresses"},
+									Item: &resolve.Object{
+										Fields: []*resolve.Field{
+											{
+												Name: []byte("street"),
+												Value: &resolve.String{
+													Path: []string{"street"},
+												},
+											},
+										},
+									},
+								},
+							},
+						},
+					},
+					OnTypeNames: [][]byte{[]byte("User")},
+				},
+			},
+		}
+
+		expected := &resolve.Object{
+			Nullable: true,
+			Fields: []*resolve.Field{
+				{
+					Name: []byte("id"),
+					Value: &resolve.String{
+						Path: []string{"id"},
+					},
+					OnTypeNames: [][]byte{[]byte("User")},
+				},
+				{
+					Name: []byte("userInfo"),
+					Value: &resolve.Object{
+						Path: []string{"userInfo"},
+						Fields: []*resolve.Field{
+							{
+								Name: []byte("kind"),
+								Value: &resolve.String{
+									Path: []string{"kind"},
+								},
+							},
+							{
+								Name: []byte("addresses"),
+								Value: &resolve.Array{
+									Path: []string{"addresses"},
+									Item: &resolve.Object{
+										Fields: []*resolve.Field{
+											{
+												Name: []byte("zip"),
```

Comment:
Can you add a test where zip and street are objects that contain fields to merge? (double nesting)